scripts/git_changes_utils.py

```diff
@@ -0,0 +1,306 @@
+# Copyright 2024 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utils file for dealing with git changes."""
+
+from __future__ import annotations
+
+import collections
+import pprint
+import subprocess
+import sys
+
+from scripts import common
+
+from typing import Dict, List, Optional, Tuple
+
+
+GitRef = collections.namedtuple(
+    'GitRef', ['local_ref', 'local_sha1', 'remote_ref', 'remote_sha1'])
+FileDiff = collections.namedtuple('FileDiff', ['status', 'name'])
+
+
+def start_subprocess_for_result(cmd: List[str]) -> Tuple[bytes, bytes]:
+    """Starts subprocess and returns (stdout, stderr)."""
+    task = subprocess.Popen(
+        cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    out, err = task.communicate()
+    return out, err
```

Comment:
Don't we already have a code for this in utils.py or similar file?

Reply:
I suppose there is common.run_cmd, but that uses check_output, which is hard to print the exact error, it is in the pre_push_hook code though; I suppose we can move this function to common?

Reply:
Yep, do so please.

Reply:
Done.